src/edges.cpp

```diff
@@ -1,8 +1,44 @@
 #include "edges.h"
+#include <map>
 
 Eigen::MatrixXi edges(const Eigen::MatrixXi &F)
 {
-  Eigen::MatrixXi E;
-  // ADD YOUR CODE HERE
-  return E;
+	Eigen::MatrixXi E;
+	// ADD YOUR CODE HERE
+
+	//Create a hashmap of unique edges 
+	typedef std::pair<int, int> Edge;
+	std::map<Edge, int> edgeMap;
+
+	//Loop through all faces, and add sorted index edges as keys to the hashmap.
+	//This will result in only unique edges being stored.
+	for (int x = 0; x < F.rows(); x++) {
+		for (int y = 0; y < F.cols(); y++) {
+
+			Edge edge(F(x, y), F(x, (y + 1) % 3));
+
+			if (F(x, (y + 1) % 3) < F(x, y)) {
+				edge.swap(Edge(F(x, (y + 1) % 3), F(x, y)));
```

Comment:
With clang, this line doesn't compile:

```
/Users/ajx/Dropbox/geometry-processing/geometry-processing-introduction/src/edges.cpp:21:10: error: too many arguments to function call, expected 1, have 2; did you mean 'std::swap'?
                                edge.swap(edge.first,edge.second);
                                     ^~~~
                                     std::swap
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:3612:1: note: 'std::swap' declared here
swap(_Tp& __x, _Tp& __y) _NOEXCEPT_(is_nothrow_move_constructible<_Tp>::value &&
^
1 error generated.
```

Easily fixed by replacing it with:

    kjkjstd::swap(edge.first,edge.second);